packages/core/helper-plugin/lib/src/hooks/useFieldHint/index.js

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { useIntl } from 'react-intl';
+
+/**
+ * @description
+ * A hook for generating the hint for a field
+ * @param {Object} description - the description of the field
+ * @param {Number} minimum - the minimum length or value of the field
+ * @param {Number} maximum - the maximum length or value of the field
+ * @param {String} units
+ */
+const useFieldHint = ({ description, minimum, maximum, units }) => {
+  const { formatMessage } = useIntl();
+
+  const [hint, setHint] = useState([]);
+
+  /**
+   * @returns {String}
+   */
+  const buildDescription = useCallback(
+    (desc) =>
+      desc?.id
+        ? formatMessage({ id: desc.id, defaultMessage: desc.defaultMessage }, { ...desc.values })
+        : '',
+    [formatMessage]
+  );
+
+  /**
+   * Constructs a suitable description of a field's minimum and maximum limits
+   * @param {Number} minimum - the minimum length or value of the field
+   * @param {Number} maximum - the maximum length or value of the field
+   * @param {String} units
+   * @returns {Array}
+   */
+  const buildMinMaxHint = useCallback(
+    (min, max, units) => {
+      const minIsNumber = typeof min === 'number';
+      const maxIsNumber = typeof max === 'number';
+
+      if (!minIsNumber && !maxIsNumber) {
+        return [];
+      }
+      const minMaxDescription = [];
+
+      if (minIsNumber) {
+        minMaxDescription.push(
+          formatMessage(
+            {
+              id: 'content-manager.form.Input.minimum',
+              defaultMessage: 'min. {min}',
+            },
+            {
+              min,
+            }
+          )
+        );
+      }
+      if (minIsNumber && maxIsNumber) {
+        const connector = ' / ';
+        minMaxDescription.push(
+          formatMessage({
+            id: connector,
+            defaultMessage: connector,
+          })
+        );
+      }
+      if (maxIsNumber) {
+        minMaxDescription.push(
+          formatMessage(
+            {
+              id: 'content-manager.form.Input.maximum',
+              defaultMessage: 'max. {max}',
+            },
+            {
+              max,
+            }
+          )
+        );
+      }
+      minMaxDescription.push(
+        formatMessage(
+          {
+            id: 'content-manager.form.Input.units',
+            defaultMessage: ' {units}{br}',
+          },
+          {
+            units,
+            br: <br />,
+          }
+        )
+      );
+
+      return minMaxDescription;
```

Comment:
I've spend some time reading the docs today and I think we might be able to combine all this in a single translation string using the select format: https://formatjs.io/docs/core-concepts/icu-syntax#select-format.

Let me know what you think:

```js
formatMessage({
  id: "something",
  defaultMessage: "{min, select, undefined {} other {min. {min}}} {max, select, undefined {} other {max. {max} }} {unit}"
},
{
  min: undefined,
  max: 100,
  hasMinAndMax: false,
  unit: formatMessage({
    id: "something.else",
    defaultMessage: "characters"
  }),
  description: ""
})
```

With this you end up having 3 translation strings:

- 1 for min/max
- 1 for the divider (/)
- 1 for unit

I think I'd like that, but curious to hear what you think.

Reply:
Hey, yes I like that idea thanks! I've spent some time this morning refactoring the code and adding this translation approach.

My only concern is that makes for quite a long translation string with a lot of conditions that might make it difficult for users. WDYT?

Reply:
I see what you mean, but I think it is fine because we make the most out a very documented translation system. If we slit it up we are more opinionated e.g. about the order of the operator which might be different in other languages.

Reply:
@joshuaellis What do you think?

Reply:
I didn't originally realise that this is what the function did (bad me) IMO this should follow the current paradigm of strings being translated (which it nearly does) but Gustav is right, we're still deciding on word order etc. which works well for English but probably wouldn't for German or French.